motupydownload/motupydownload.py

```diff
@@ -122,34 +165,24 @@
     MOTU_USER = os.environ.get("MOTU_USER", "NONE")
     MOTU_PASSWORD = os.environ.get("MOTU_PASSWORD", "NONE")
 
-    # call motuclient for every variable and write into daily output file
-    for variable_name in variables:
-        for day in range(num_days):
-            # set time limits corresponding to start_time and end_time in string format
-            start_time = (time_start + timedelta(day)).strftime("%Y-%m-%d")
-            end_time = (time_start + timedelta(day + 1)).strftime("%Y-%m-%d")
-            # set time limits for motuclient in format "YYYY-mm-dd ss:MM:HH"
-            time_left_motu = f"{start_time}T00:00:00"
-            time_right_motu = f"{end_time}T00:00:00"
+    num_cores = multiprocessing.cpu_count()
```

Comment:
I'd rather use a command line flag for setting the number of parallel downloads:
- We're not really using CPU anyway (time is determined by the download bandwidht).
- On large systems like the Nesh frontends, `cpu_count()` might be very large (48 or 64 or 96). This is probably higher than the number of parallel downloads allowed by the MOTU server.

Reply:
Thank you for comments, last week you showed me one way to parallelize the loop, could you send me the link to look it up again.

Reply:
I've suggested to use `multiprocessing.Pool.map` as described here: https://docs.python.org/3/library/multiprocessing.html The advantage of using `multiprocessing`, which is part of the Python standard library, would be avoiding an additional dependency.